packages/template-retail-react-app/app/pages/product-detail/index.jsx

```diff
@@ -117,22 +158,48 @@ const ProductDetail = ({category, product, isLoading}) => {
             status: 'error'
         })
     }
+
+    const addItemToBasket = (basketId, variant, quantity) => {
+        const productItems = [
+            {
+                productId: variant.productId,
+                quantity,
+                price: variant.price
+            }
+        ]
+
+        addItemToBasketAction.mutate(
+            {parameters: {basketId}, body: productItems},
+            {
+                onSuccess: () => {
+                    // only show this modal when a product is successfully add to cart
+                    onAddToCartModalOpen({product, quantity})
+                },
+                onError: (err) => {
+                    showError(err)
```

Comment:
Minor bug that exists from before: `showError` actually does not accept the `err` argument.

Reply:
ah right, we did have a discussion about this issue, we chose to go with the generic message so it could be translated. I will not pass err in this case then, we can change it after we finalize how we want to show error message